Remove 1.21.x testing

kfox1111 opened the pull request:
Other tests now are depending on 1.22+.

faisal-memon commented:
Thanks @kfox1111. What do you think of removing 1.22-1.24 as well to get within the n-2 range?

faisal-memon commented:
Can you update the base README.md and remove k8s 1.21 from there?

kfox1111 commented:
I do think some of the integration tests we do are somewhat excessive and could be handled better with unit testing instead. That's where something like the unit test framework in https://github.com/spiffe/helm-charts/pull/260 could come in handy. We could then remove some of the integration tests reducing the number of permutations that way?

kfox1111 commented:
> Also need to make the same change to `helm-chart-ci-ignore.yaml`

I looked for it there but couldn't find it?

faisal-memon commented:
> > Also need to make the same change to `helm-chart-ci-ignore.yaml`
> 
> I looked for it there but couldn't find it?

Yes, sorry, not needed there.

faisal-memon commented on charts/spire/README.md.gotmpl:
```diff
@@ -21,9 +21,9 @@
+| Kubernetes | `1.22+`            |
 
-> **Note**: For Kubernetes, we will officially support the last 3 versions as described in [k8s versioning](https://kubernetes.io/releases/version-skew-policy/#supported-versions). Any version before the last 3 we will try to support as long it doesn't bring security issues or any big maintenance burden. *The first version we tested this chart with is `1.21`.*
+> **Note**: For Kubernetes, we will officially support the last 3 versions as described in [k8s versioning](https://kubernetes.io/releases/version-skew-policy/#supported-versions). Any version before the last 3 we will try to support as long it doesn't bring security issues or any big maintenance burden. *The first version we tested this chart with is `1.22`.*
```
```suggestion
> **Note**: For Kubernetes, we will officially support the last 3 versions as described in [k8s versioning](https://kubernetes.io/releases/version-skew-policy/#supported-versions). Any version before the last 3 we will try to support as long it doesn't bring security issues or any big maintenance burden. 
```

faisal-memon replied:
Dont think we need that bit of history anymore.

faisal-memon reviewed (approved):
Thanks @kfox1111

edwbuck reviewed (changes requested):
What about the generated README.md in the root of the repo?

What about the kubeVersion in charts/spire/Chart.yaml?

kfox1111 commented:
> What about the generated README.md in the root of the repo?

Fixed
 
> What about the kubeVersion in charts/spire/Chart.yaml?

Fixed. It needs to stay that way to not block installs. But I was able to fix up the docs.

edwbuck reviewed (approved):
With the last two commits, all docs now properly report k8s versions 1.22+ are supported, and the Charts.yaml will still deploy on 1.21+, for those that have 1.21 clusters and desire to take their own risks in running on something we might not fully test.

Approved
